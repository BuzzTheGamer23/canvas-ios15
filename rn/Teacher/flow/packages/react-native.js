// @flow
export type PanResponderGestureState = {
  stateID: number,
  moveX: number,
  moveY: number,
  x0: number,
  y0: number,
  dx: number,
  dy: number,
  vx: number,
  vy: number,
}
