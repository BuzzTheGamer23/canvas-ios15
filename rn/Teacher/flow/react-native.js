//
// Copyright (C) 2017-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

// @flow
/* eslint-disable */

// define basic style prop types
// the internal ones are the best we have
import {
  type ____ViewStyleProp_Internal,
  type ____TextStyleProp_Internal,
  type ____ImageStyleProp_Internal,
} from 'react-native/Libraries/StyleSheet/StyleSheetTypes'

declare type ViewStyleProp = ____ViewStyleProp_Internal
declare type TextStyleProp = ____TextStyleProp_Internal
declare type ImageStyleProp = ____ImageStyleProp_Internal
