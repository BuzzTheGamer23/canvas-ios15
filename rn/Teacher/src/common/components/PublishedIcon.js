//
// Copyright (C) 2016-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

/**
* @flow
*/

import React from 'react'
import {
  View,
  Image,
  StyleSheet,
} from 'react-native'

import Images from '../../images'
import i18n from 'format-message'

type Props = {
  published: true,
  tintColor: string,
  image: any,
}

export default class PublishedIcon extends React.Component<any, Props, any> {
  render () {
    const published = this.props.published
    const icon = published ? Images.published : Images.unpublished
    const iconStyle = published ? styles.publishedIcon : styles.unpublishedIcon
    const accessibilityLabel = published ? i18n('Published') : i18n('Not Published')
    return (
      <View style={styles.container} accessibilityLabel={accessibilityLabel}>
        <Image source={this.props.image} style={[styles.image, { tintColor: this.props.tintColor }]} />
        <View style={styles.publishedIconContainer}>
          <Image source={icon} style={iconStyle} />
        </View>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 0,
    height: 32,
    width: 46,
    alignItems: 'center',
  },
  image: {
    position: 'absolute',
  },
  publishedIconContainer: {
    justifyContent: 'center',
    alignItems: 'center',
    position: 'absolute',
    top: 10,
    left: 20,
    backgroundColor: 'white',
    height: 16,
    width: 16,
    borderRadius: 8,
  },
  publishedIcon: {
    height: 12,
    width: 12,
    tintColor: '#00AC18',
  },
  unpublishedIcon: {
    height: 12,
    width: 12,
    tintColor: '#8B969E',
  },
})
