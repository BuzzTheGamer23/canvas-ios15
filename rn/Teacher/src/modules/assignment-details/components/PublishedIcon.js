/**
 * @flow
 */

import React from 'react'
import { View, Image, StyleSheet } from 'react-native'
import Images from '../../../images'
import i18n from 'format-message'
import { Text } from '../../../common/text'
import colors from '../../../common/colors'

export default class PublishedIcon extends React.Component {
  render (): React.Element<*> {
    let { published, style, iconSize } = this.props

    const publishedIcon = published ? Images.assignments.published : Images.assignments.unpublished

    let iconStyle = published ? internalStyle.publishedIcon : internalStyle.unpublishedIcon
    let customStyle = {}
    if (iconSize) {
      customStyle = { width: iconSize, height: iconSize }
    }

    let unpublishedText = i18n({
      default: 'Unpublished',
      description: 'Unpublished text',
    })

    let publishedText = i18n({
      default: 'Published',
      description: 'Published text',
    })

    return (
      <View style={[ style, internalStyle.container ]}>
        <Image source={publishedIcon} style={[iconStyle, customStyle]}/>
        <Text style={[internalStyle.text, published ? internalStyle.publishedTextStyle : internalStyle.unPublishedTextStyle]}>{(published) ? publishedText : unpublishedText}</Text>
      </View>
    )
  }
}

const iconSize = 21

const internalStyle = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'row',
    alignItems: 'center',
  },
  text: {
    marginLeft: 5,
    fontWeight: '500',
    fontSize: 16,
    color: 'grey',
  },
  publishedTextStyle: { color: colors.checkmarkGreen },
  unPublishedTextStyle: { color: 'grey' },
  publishedIcon: {
    height: iconSize,
    width: iconSize,
    tintColor: colors.checkmarkGreen,
  },
  unpublishedIcon: {
    height: iconSize,
    width: iconSize,
    tintColor: 'grey',
  },
})
