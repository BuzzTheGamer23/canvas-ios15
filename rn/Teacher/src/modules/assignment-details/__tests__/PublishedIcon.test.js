/**
 * @flow
 */

import 'react-native'
import React from 'react'
import PublishedIcon from '../components/PublishedIcon'
import renderer from 'react-test-renderer'

test('render published', () => {
  let tree = renderer.create(
    <PublishedIcon published={true} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('render unpublished', () => {
  let tree = renderer.create(
    <PublishedIcon published={false} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('render larger icon size', () => {
  let tree = renderer.create(
    <PublishedIcon published={true} iconSize={75} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
