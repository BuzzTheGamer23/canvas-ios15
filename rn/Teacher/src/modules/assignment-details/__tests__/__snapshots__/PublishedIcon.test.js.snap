// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render larger icon size 1`] = `
<View
  style={
    Array [
      undefined,
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
      },
    ]
  }
>
  <Image
    source={1}
    style={
      Array [
        Object {
          "height": 21,
          "tintColor": "#00AC18",
          "width": 21,
        },
        Object {
          "height": 75,
          "width": 75,
        },
      ]
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {},
        Array [
          Object {
            "color": "grey",
            "fontSize": 16,
            "fontWeight": "500",
            "marginLeft": 5,
          },
          Object {
            "color": "#00AC18",
          },
        ],
      ]
    }
  >
    Published
  </Text>
</View>
`;

exports[`render published 1`] = `
<View
  style={
    Array [
      undefined,
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
      },
    ]
  }
>
  <Image
    source={1}
    style={
      Array [
        Object {
          "height": 21,
          "tintColor": "#00AC18",
          "width": 21,
        },
        Object {},
      ]
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {},
        Array [
          Object {
            "color": "grey",
            "fontSize": 16,
            "fontWeight": "500",
            "marginLeft": 5,
          },
          Object {
            "color": "#00AC18",
          },
        ],
      ]
    }
  >
    Published
  </Text>
</View>
`;

exports[`render unpublished 1`] = `
<View
  style={
    Array [
      undefined,
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
      },
    ]
  }
>
  <Image
    source={1}
    style={
      Array [
        Object {
          "height": 21,
          "tintColor": "grey",
          "width": 21,
        },
        Object {},
      ]
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {},
        Array [
          Object {
            "color": "grey",
            "fontSize": 16,
            "fontWeight": "500",
            "marginLeft": 5,
          },
          Object {
            "color": "grey",
          },
        ],
      ]
    }
  >
    Unpublished
  </Text>
</View>
`;
